### PR TITLE
Factor out and test point evaluation

### PR DESCRIPTION
**Pull request by jonsyu1**
Point evaluation is tricky and complex. Factor it out to a separate file and test the obvious cases.

Fixes: #94

### CODE REVIEW

File: server/play/model/GameEvaluation.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
+import { filter, isEqual, remove } from "lodash";
+import { isBout, Bout, Card, Suit, TheJoker, TheOne, TrumpValue } from "./Card";
+import { cardsContain, getCardPoint } from './CardUtils';
+import { BidValue, Call, CompletedTrick, JokerExchangeState, Outcome, PlayerId, ShowTrumpState } from "./GameState";
+
+export interface Earnings {
```
Comment (jonsyu1) [inline server/play/model/GameEvaluation.ts:6]: These were pulled from game state reducer. I will indicate the changes I made.

File: server/play/model/GameEvaluation.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        if (cardsLost.length > 0) {
+          remove(cardsWon, (card) => isEqual(card, TheJoker));
+          // trade 4.5 point card for half point card
+          pointsEarned -= 4;
+        }
+      }
+    }
+  }
+  const bouts = filter(cardsWon, (card): card is Bout => isBout(card));
+  bouts.push(...filter(dog, (card): card is Bout => isBout(card)))
```
Comment (jonsyu1) [inline server/play/model/GameEvaluation.ts:43]: I changed the bouts to include bouts in the dog.

File: server/play/model/GameEvaluation.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+          pointsEarned -= 4;
+        }
+      }
+    }
+  }
+  const bouts = filter(cardsWon, (card): card is Bout => isBout(card));
+  const dogPoints = dog.map(getCardPoint).reduce((a, b) => a + b, 0);
+  if (bid !== BidValue.ONESIXTY) {
+    pointsEarned += dogPoints;
+    bouts.push(...filter(dog, (card): card is Bout => isBout(card)))
```
Comment (jonsyu1) [inline server/play/model/GameEvaluation.ts:46]: This now includes bouts in the dog.

**Comment by jonsyu1** [comment]
Fixes: #94

**Review by gillessed — APPROVED** [review]

**Pull request merged**